nsxt/resource_nsxt_policy_context_profile.go

```diff
@@ -0,0 +1,508 @@
+	d.Set("path", obj.Path)
+	d.Set("revision", obj.Revision)
+	fillAttributesInSchema(d, obj.Attributes)
+
+	return nil
+}
+
+func resourceNsxtPolicyContextProfileUpdate(d *schema.ResourceData, m interface{}) error {
+	connector := getPolicyConnector(m)
+	id := d.Id()
+	if id == "" {
+		return fmt.Errorf("Error obtaining ContextProfile ID")
+	}
+
+	// Read the rest of the configured parameters
+	displayName := d.Get("display_name").(string)
+	description := d.Get("description").(string)
+	attributesStructList := make([]model.PolicyAttributes, 0, 0)
+	var err error
+	for key := range attributeKeyMap {
+		attributes := d.Get(key).(*schema.Set).List()
+		err := checkAttributesValid(attributes, m, key)
+		if err != nil {
+			return err
+		}
+		attributeStructList, err := constructAttributesModelList(attributes, key)
+		if err != nil {
+			return err
+		}
+		attributesStructList = append(attributesStructList, attributeStructList...)
+	}
+	tags := getPolicyTagsFromSchema(d)
+
+	obj := model.PolicyContextProfile{
+		DisplayName: &displayName,
+		Description: &description,
+		Tags:        tags,
+		Attributes:  attributesStructList,
+	}
+
+	// Update the resource using PATCH
+	if isPolicyGlobalManager(m) {
+		rawObj, err1 := convertModelBindingType(obj, model.PolicyContextProfileBindingType(), gm_model.PolicyContextProfileBindingType())
+		if err1 != nil {
+			return err1
+		}
+		gmObj := rawObj.(gm_model.PolicyContextProfile)
+		client := gm_infra.NewDefaultContextProfilesClient(connector)
+		err = client.Patch(id, gmObj)
+	} else {
+		client := infra.NewDefaultContextProfilesClient(connector)
+		err = client.Patch(id, obj)
+	}
+
+	if err != nil {
+		return handleUpdateError("ContextProfile", id, err)
+	}
+
+	return resourceNsxtPolicyContextProfileRead(d, m)
+}
+
+func resourceNsxtPolicyContextProfileDelete(d *schema.ResourceData, m interface{}) error {
+	id := d.Id()
+	if id == "" {
+		return fmt.Errorf("Error obtaining ContextProfile ID")
+	}
+
+	connector := getPolicyConnector(m)
+	var err error
+	force := true
+	if isPolicyGlobalManager(m) {
+		client := gm_infra.NewDefaultContextProfilesClient(connector)
+		err = client.Delete(id, &force)
+	} else {
+		client := infra.NewDefaultContextProfilesClient(connector)
+		err = client.Delete(id, &force)
+	}
+	if err != nil {
+		return handleDeleteError("ContextProfile", id, err)
+	}
+
+	return nil
+}
+
+func checkAttributesValid(attributes []interface{}, m interface{}, key string) error {
+
+	if key == "app_id" {
+		err := validateSubAttributes(attributes)
+		if err != nil {
+			return err
+		}
+	}
+	var attrClient interface{}
+	connector := getPolicyConnector(m)
+	isPolicyGlobalManager := isPolicyGlobalManager(m)
+	if isPolicyGlobalManager {
+		attrClient = gm_cont_prof.NewDefaultAttributesClient(connector)
+	} else {
+		attrClient = cont_prof.NewDefaultAttributesClient(connector)
+	}
+	attributeValues, err := listAttributesWithKey(attributeKeyMap[key], attrClient, isPolicyGlobalManager)
```

Comment:
I wonder if it would make sense to cache this. But I don't expect large amounts of context profiles to be created, so probably an overkill